source/postprocess/visualization/volume_of_fluid_values.cc

```diff
@@ -0,0 +1,201 @@
+      VolumeOfFluidValues<dim>::
+      get_data_component_interpretation () const
+      {
+        return interpretations;
+      }
+
+
+      template <int dim>
+      UpdateFlags
+      VolumeOfFluidValues<dim>::
+      get_needed_update_flags () const
+      {
+        return update_values;
+      }
+
+
+      template <int dim>
+      void
+      VolumeOfFluidValues<dim>::
+      evaluate_vector_field(const DataPostprocessorInputs::Vector<dim> &input_data,
+                            std::vector<Vector<double>> &computed_quantities) const
+      {
+        const unsigned int n_quadrature_points = input_data.solution_values.size();
+        Assert (computed_quantities.size() == n_quadrature_points, ExcInternalError ());
+        Assert (input_data.solution_values[0].size() == this->introspection().n_components, ExcInternalError ());
+
+        unsigned int out_per_field=1;
+        if (include_contour)
+          out_per_field += 1;
+        if (include_normal)
+          out_per_field += dim;
+
+        for (unsigned int f=0; f<this->get_volume_of_fluid_handler().get_n_fields(); ++f)
+          {
+            VolumeOfFluidField<dim> field = this->get_volume_of_fluid_handler().field_struct_for_field_index(f);
+
+            const FEVariable<dim> &volume_of_fluid_var = field.volume_fraction;
+            const unsigned int volume_of_fluid_ind = volume_of_fluid_var.first_component_index;
+            const FEVariable<dim> &volume_of_fluidLS_var = field.level_set;
+            const unsigned int volume_of_fluidLS_ind = volume_of_fluidLS_var.first_component_index;
+
+            for (unsigned int q=0; q<n_quadrature_points; ++q)
+              {
+                unsigned int out_ind = f*out_per_field;
+                computed_quantities[q][out_ind] = input_data.solution_values[q][volume_of_fluid_ind];
+                ++out_ind;
+                if (include_contour)
+                  {
+                    computed_quantities[q][out_ind] = input_data.solution_values[q][volume_of_fluidLS_ind];
+                    ++out_ind;
+                  }
+
+                if (include_normal)
+                  {
+                    Tensor<1, dim, double> normal = -input_data.solution_gradients[q][volume_of_fluidLS_ind];
+                    for (unsigned int i = 0; i<dim; ++i)
+                      {
+                        computed_quantities[q][out_ind] = normal[i];
+                        ++out_ind;
+                      }
+                  }
+              }
+          }
+      }
+
+
+      template <int dim>
+      void
+      VolumeOfFluidValues<dim>::declare_parameters (ParameterHandler &prm)
+      {
+        prm.enter_subsection("Postprocess");
+        {
+          prm.enter_subsection("Visualization");
+          {
+            prm.enter_subsection("Volume of Fluid");
+            {
+              prm.declare_entry("Include internal reconstruction contour", "false",
```

Comment:
I find the term `include` is not quite clear. What about `Output internal reconstruction contour`?